cdap-app-fabric/src/main/java/io/cdap/cdap/app/preview/PreviewRunnerModule.java

```diff
@@ -91,73 +88,50 @@ public class PreviewRunnerModule extends PrivateModule {
-      bind(ArtifactRepository.class)
-        .annotatedWith(Names.named(AppFabricServiceRuntimeModule.NOAUTH_ARTIFACT_REPO))
-        .to(RemoteArtifactRepositoryWithLocalization.class)
-        .in(Scopes.SINGLETON);
-      expose(ArtifactRepository.class).annotatedWith(Names.named(AppFabricServiceRuntimeModule.NOAUTH_ARTIFACT_REPO));
-
-
-      // Use remote implementation to fetch plugin metadata from AppFab.
-      // Remote implementation internally uses artifact localizer to fetch and cache artifacts locally.
-      bind(PluginFinder.class).to(RemoteWorkerPluginFinder.class);
-      expose(PluginFinder.class);
-
-      // Use remote implementation to fetch preferences from AppFab.
-      bind(PreferencesFetcher.class).to(RemotePreferencesFetcherInternal.class);
-      expose(PreferencesFetcher.class);
-    } else {
-      bind(ArtifactRepositoryReader.class).toProvider(artifactRepositoryReaderProvider);
-      bind(ArtifactRepository.class).to(DefaultArtifactRepository.class);
-      expose(ArtifactRepository.class);
-
-      bind(ArtifactRepository.class)
-        .annotatedWith(Names.named(AppFabricServiceRuntimeModule.NOAUTH_ARTIFACT_REPO))
-        .to(DefaultArtifactRepository.class)
-        .in(Scopes.SINGLETON);
-      expose(ArtifactRepository.class)
-        .annotatedWith(Names.named(AppFabricServiceRuntimeModule.NOAUTH_ARTIFACT_REPO));
-
-      bind(PluginFinder.class).toProvider(pluginFinderProvider);
-      expose(PluginFinder.class);
-
-      bind(PreferencesFetcher.class).toProvider(preferencesFetcherProvider);
-      expose(PreferencesFetcher.class);
-    }
+    // Use remote implementation to fetch artifact metadata from AppFab.
+    // Remote implementation internally uses artifact localizer to fetch and cache artifacts locally.
+    bind(ArtifactRepositoryReader.class).to(RemoteArtifactRepositoryReaderWithLocalization.class);
+    bind(ArtifactRepository.class).to(RemoteArtifactRepositoryWithLocalization.class);
+    expose(ArtifactRepository.class);
+    bind(ArtifactRepository.class)
+      .annotatedWith(Names.named(AppFabricServiceRuntimeModule.NOAUTH_ARTIFACT_REPO))
+      .to(RemoteArtifactRepositoryWithLocalization.class)
+      .in(Scopes.SINGLETON);
+    expose(ArtifactRepository.class).annotatedWith(Names.named(AppFabricServiceRuntimeModule.NOAUTH_ARTIFACT_REPO));
+
+
+    // Use remote implementation to fetch plugin metadata from AppFab.
+    // Remote implementation internally uses artifact localizer to fetch and cache artifacts locally.
+    bind(PluginFinder.class).to(RemoteWorkerPluginFinder.class);
+    expose(PluginFinder.class);
+
+    bind(PreferencesFetcher.class).to(RemotePreferencesFetcherInternal.class);
+    expose(PreferencesFetcher.class);
+
+    // Read artifact locations from disk when using artifact localizer due to shared mounted read-only PD.
+    install(new LocalLocationModule());
+    expose(LocationFactory.class);
+
+    // Use remote implementation to fetch preferences from AppFab.
```

Comment:
duplicate bindings from above?

Reply:
Can you clarify where the duplicates are? I only see one `LocalLocationModule` in the PR.

Reply:
line 125

Reply:
Fixed, thanks!